onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py

```diff
@@ -247,9 +247,6 @@ def _where_input_wrangler(
     "acosh": core_ops.aten_acosh,
     "add": core_ops.aten_add,
     "addmm": core_ops.aten_addmm,
-    "arange_start_step": core_ops.aten_arange_start_step,
```

Comment:
The issue was not raised by unit tests earlier. Is it because we are using reference evaluator? Is it a bug in reference evaluator that it overlooks the if-branch-same-dtype requirement?

Reply:
onnx.checker.check_function missed to check it I think.

Reply:
Filed https://github.com/microsoft/onnx-script/issues/423

Reply:
the evaluator actually executes line by line so its behavior may be different when executing a function. @gramalingam should we allow a function mode? e.g. `ORTEvaluator(compile_functions=True)` But that will not allow break points any more...

Reply:
We do run both the reference and ort in test.

Reply:
Yes, as Justin says, execution of control-flow constructs uses python (for better debugging) and will not invoke the ONNX checker (for an IF or LOOP). 

One possibility is to add a checker API, that looks like a normal eager-mode call, but invokes the checker (with an appropriate context). Sort of similar to Justin's suggestion above, except it is just a checker with no-execution.

Reply:
Yes I don't see that as mutually exclusive (do checker first, then eager execute). Maybe a performance concern?

Reply:
Agree. The infrastructure should enable the users to decide whether they want separate tests or always run the checker in eager-mode, etc.